lib/meandro/rules/unused_struct_field.ex

```diff
@@ -0,0 +1,155 @@
+defmodule Meandro.Rule.UnusedStructField do
+  @moduledoc """
+  Finds struct fields that are not used.
+  It has the following assumptions:
+    - As we are not tracking the value of each variable and a variable can be
+      reassigned we don't check the struct name when looking for access to a
+      struct field or modification
+  """
+
+  @behaviour Meandro.Rule
+
+  @impl Meandro.Rule
+  def analyze(files_and_asts, _options) do
+    List.flatten(
+      for {file, module_asts} <- files_and_asts,
+          {_module_name, ast} <- module_asts,
+          result <- analyze_module(file, ast, files_and_asts) do
+        result
+      end
+    )
+  end
+
+  @impl Meandro.Rule
+  def is_ignored?({struct, field}, {struct, field}) do
+    true
+  end
+
+  def is_ignored?({struct, _field}, struct) do
+    true
+  end
+
+  def is_ignored?(_pattern, _ignore_spec) do
+    false
+  end
+
+  defp analyze_module(file, ast, files_and_asts) do
+    struct_info = struct_info(ast)
+
+    case Map.get(struct_info, :fields) do
+      nil ->
+        []
+
+      fields ->
+        module_name = struct_info[:module_name]
+        module_aliases = struct_info[:module_aliases]
+
+        for field <- fields do
+          unused = is_unused?({field, module_name, module_aliases}, files_and_asts)
+
+          case unused do
+            true ->
+              %Meandro.Rule{
+                file: file,
+                rule: __MODULE__,
+                pattern: {module_name, field},
+                text: "The field #{field} from the struct #{module_name} is unused"
+              }
+
+            false ->
+              []
```

Comment:
The true branch returns a struct and this one a list. Shouldn't either both return lists or this one return `nil`, instead?